### PR TITLE
ModernEventSystem: fix event replaying

### PR DESCRIPTION
**Pull request by trueadm**
If you run all the tests with modern event system flag:

```
export const enableModernEventSystem = true;
```

This PR, thanks to Seb now should fix those use-cases with the modern event system and the event replaying mechanics.

### CODE REVIEW

**Comment by trueadm** [comment]
After speaking to @sebmarkbage it sounds like we need a larger fix.

File: packages/react-dom/src/events/DOMModernPluginEventSystem.js [hunk 1]
```diff
@@ -302,60 +302,62 @@ export function dispatchEventForPluginEventSystem(
...
-    // systems is that the modern event system now has an event listener
-    // attached to each React Root and React Portal Root. Together,
-    // the DOM nodes representing these roots are the "rootContainer".
-    // To figure out which ancestor instance we should use, we traverse
-    // up the fiber tree from the target instance and attempt to find
-    // root boundaries that match that of our current "rootContainer".
-    // If we find that "rootContainer", we find the parent fiber
-    // sub-tree for that root and make that our ancestor instance.
-    let node = targetInst;
+    if (targetInst !== null) {
```
Comment (sebmarkbage) [inline packages/react-dom/src/events/DOMModernPluginEventSystem.js]: Why was this needed?

Comment (trueadm) [reply packages/react-dom/src/events/DOMModernPluginEventSystem.js]: The event replaying system passed `null` for target fiber. If we don't guard against it we will just early return and never dispatch anything, which can cause issues if I'm not mistaken?
